jack_bot.py

```diff
@@ -188,6 +150,14 @@ def can_train(self, unit_type, larva=True):
         """Global requirements for creating an unit"""
         return (not larva or self.larvae) and self.can_afford(unit_type) and self.can_feed(unit_type)
 
+    def can_build_uniques(self, unit_type, building):
```

Comment:
i would suggest rename it to `can_build_unique` since you it checks only one not multiple

Reply:
done